internal/services/compute/image_resource.go

```diff
@@ -463,7 +476,8 @@ func flattenImageOSDisk(input *images.ImageStorageProfile) []interface{} {
 			}
 			diskEncryptionSetId := ""
 			if set := v.DiskEncryptionSet; set != nil && set.Id != nil {
-				diskEncryptionSetId = *set.Id
+				encryptionId, _ := commonids.ParseDiskEncryptionSetIDInsensitively(*set.Id)
+				diskEncryptionSetId = encryptionId.String()
```

Comment:
`.String()` outputs the ID information in a human-readable way for error and log messages, these probably want to use `.ID()`
```suggestion
				diskEncryptionSetId = encryptionId.ID()
```